src/Raven.Server/Documents/Subscriptions/Processor/DocumentsDatabaseSubscriptionProcessor.cs

```diff
@@ -189,6 +175,20 @@ protected override SubscriptionBatchItem ShouldSend(Document item, out string re
                 };
             }
 
+            if (item.Flags.Contain(DocumentFlags.Archived) && SubscriptionState.ArchivedDataProcessingBehavior == ArchivedDataProcessingBehavior.ExcludeArchived)
```

Comment:
we need to use `result.Document` instead of `item`, since we load the new document from storage if we are currently fetching from the resend list.

also make sure to assign the new Flags (add something like `Flags = current.Flags` on line 174)

Reply:
done, thanks, saved me a bug :)